python/tests/src/test_access_token_common.py

```diff
@@ -0,0 +1,68 @@
+import unittest
+import mock
+import json
+
+from src.access_token_common import AccessTokenCommon
+
+class AccessTokenCommonTest(unittest.TestCase):
+
+    mock_os_environ_access_token = {'ACCESS_TOKEN_FROM_ENV': 'access token 42'}
+    @mock.patch.dict('os.environ', mock_os_environ_access_token, clear=True)
+    def test_access_token_from_environment(self):
+        mock_api_config = mock.Mock()
+        mock_api_config.app_id = 'test-app-id'
+        mock_api_config.app_secret = 'test-app-secret'
+        mock_api_config.oauth_token_dir = 'test-token-dir'
+
+        # The name of the access token is converted to upper case to find
+        # it in the process environment.
+        access_token = AccessTokenCommon(mock_api_config, name='access_token_from_env')
+        access_token.fetch()
+        # echo -n 'access token 42' | shasum -a 256
+        self.assertEqual(access_token.hashed(),
+                         '553c1f363497ba07fecc989425e57e37c2b5f57ff7476c79dfd580ef0741db88')
+        with self.assertRaisesRegex(RuntimeError, 'AccessToken does not have a refresh token'):
+            access_token.hashed_refresh_token()
+
+    mock_os_environ_empty = {}
+    @mock.patch.dict('os.environ', mock_os_environ_empty, clear=True)
+    @mock.patch('os.chmod')
+    @mock.patch('builtins.open')
```

Comment:
Consider using [`mock.mock_open`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.mock_open).

Reply:
I looked into mock_open for about 45 minutes. In this context, mock_open seems to add more complexity, because:
* mock_open seems to expect the read_data to be relatively simple and specified when the mock is instantiated
* It's tricky to verify the fileno used by mock_open against the fileno used by chmod.
* While it's possible to verify the calls to write(), it's a lot more complicated to verify each of the write calls than it is to construct a single string to verify from a number of calls to a mocked write function.

I'm pretty sure that it would be possible to work-around all of these issues, but don't think that the resulting test would be more readable.